cos-fleetshard-operator-camel/src/main/java/org/bf2/cos/fleetshard/operator/camel/CamelOperandSupport.java

```diff
@@ -373,25 +373,20 @@ public static Optional<KameletBinding> lookupBinding(KubernetesClient client, Ma
     }
 
     public static void computeStatus(ConnectorStatusSpec statusSpec, KameletBindingStatus kameletBindingStatus) {
-        if (kameletBindingStatus.phase != null) {
-            switch (kameletBindingStatus.phase.toLowerCase(Locale.US)) {
-                case KameletBindingStatus.PHASE_READY:
-                    statusSpec.setPhase(ManagedConnector.STATE_PROVISIONING);
-                    if (kameletBindingStatus.conditions != null) {
-                        boolean readyCondition = kameletBindingStatus.conditions.stream()
-                            .anyMatch(cond -> "Ready".equals(cond.getType())
-                                && "True".equals(cond.getStatus()));
-                        if (readyCondition) {
-                            statusSpec.setPhase(ManagedConnector.STATE_READY);
-                        }
+        statusSpec.setPhase(ManagedConnector.STATE_PROVISIONING);
+        if (kameletBindingStatus.conditions != null) {
+            Optional<Condition> readyConditionOpt = kameletBindingStatus.conditions.stream()
+                .filter(cond -> "Ready".equals(cond.getType())).findAny();
+            if (readyConditionOpt.isPresent()) {
+                Condition readyCondition = readyConditionOpt.get();
+                boolean klbReady = "True".equals(readyCondition.getStatus());
+                if (klbReady) {
+                    statusSpec.setPhase(ManagedConnector.STATE_READY);
+                } else {
+                    if (isErrorReasons(readyCondition.getReason())) {
+                        statusSpec.setPhase(ManagedConnector.STATE_FAILED);
                     }
-                    break;
-                case KameletBindingStatus.PHASE_ERROR:
-                    statusSpec.setPhase(ManagedConnector.STATE_FAILED);
```

Comment:
isn't this still a thing ? i.e if there is a technical error while reconciling the integration

@squakez what do you think ?

Reply:
As discussed, I completely remove dependency from KBL PHASE and move all checks on conditions only. I am hoping if there is some technical error then KBL Ready condition would be set to `false`.

Reply:
Yep but I'm not very sure that the ready condition is alway present, as example in the kamelet binding initialization reconcile action from camel-k (https://github.com/apache/camel-k/blob/d5b9466248fdcc2eedf6489cec978c6b527869ff/pkg/controller/kameletbinding/initialize.go#L52), in case of an issue creating the integration, the condition that is set is `IntegrationError` and I don't see any track of a `Ready` condition.

btw, I thin the type of thjat condition is very misleading as then you gonna have `Type: IntegrationError` and `Status: False`

Reply:
Yes, also consider that it's the canonical way we rebuild the KameletBindings (ie, wiping it off the status completely), though I am not sure if that is something that may affect you.